Compiling from git on Mac OS X error: ‘MSG_NOSIGNAL’ undeclared

Hello,

When compiling from Git on Mac OS X, I get the following error:

```
Making all in daemon
  CC       collectd-collectd.o
collectd.c: In function ‘notify_systemd’:
collectd.c:474: error: ‘MSG_NOSIGNAL’ undeclared (first use in this function)
collectd.c:474: error: (Each undeclared identifier is reported only once
collectd.c:474: error: for each function it appears in.)
```

`git blame` seems to think that this was introduced in ff270e6d53374bddca9f3fe0bb7a8c2294548edc by @pyr 

MSG_NOSIGNAL is not defined anywhere within /usr/include on Mac OS 10.8.5

nick.

--- a/src/daemon/collectd.c
+++ b/src/daemon/collectd.c
@@ -414,6 +414,7 @@ static int pidfile_remove (void)
 } /* static int pidfile_remove (const char *file) */
 #endif /* COLLECT_DAEMON */
 
+#ifdef KERNEL_LINUX
 int notify_upstart (void)
 {
     const char  *upstart_job = getenv("UPSTART_JOB");
@@ -479,6 +480,7 @@ int notify_systemd (void)
     close(fd);
     return 1;
 }
+#endif /* KERNEL_LINUX */
 
 int main (int argc, char **argv)
 {
@@ -598,9 +600,13 @@ int main (int argc, char **argv)
 
     /*
      * Only daemonize if we're not being supervised
-     * by upstart or systemd.
+     * by upstart or systemd (when using Linux).
      */
-	if (daemonize && notify_upstart() == 0 && notify_systemd() == 0)
+	if (daemonize
+#ifdef KERNEL_LINUX
+	    && notify_upstart() == 0 && notify_systemd() == 0
+#endif
+	)
 	{
 		if ((pid = fork ()) == -1)
 		{

